subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/PrecompiledScriptPluginIntegrationTest.kt

```diff
@@ -721,4 +722,49 @@ class PrecompiledScriptPluginIntegrationTest : AbstractPluginIntegrationTest() {
             assertThat(output, containsString("my-plugin settings plugin applied"))
         }
     }
+
+    @Test
+    fun `should not allow precompiled plugin to conflict with core plugin`() {
+        withKotlinBuildSrc()
+        withFile(
+            "buildSrc/src/main/kotlin/java.gradle.kts",
+            """
+            tasks.register("myTask") {}
+            """
+        )
+        withDefaultSettings()
+        withFile(
+            "build.gradle",
+            """
+            plugins {
+                java
+            }
+            """
+        )
+
+        val error = buildAndFail("help")
+
+        error.assertHasCause("Precompiled plugin: 'java.gradle.kts' conflicts with the core plugin: 'java'. " +
+            "See https://docs.gradle.org/" + GradleVersion.current().version + "/userguide/custom_plugins.html#sec:precompiled_plugins for more details.")
+    }
+
+    @Test
+    fun `should not allow precompiled plugin to have org-dot-gradle prefix`() {
+        // given:
+        withKotlinBuildSrc()
+        withFile(
+            "buildSrc/src/main/kotlin/org.gradle.my-plugin.gradle.kts",
+            """
+            tasks.register("myTask") {}
+            """
+        )
+        withDefaultSettings()
+
+        // when:
+        val error = buildAndFail("help")
+
+        // then:
+        error.assertHasCause("Precompiled plugin should not have prefix: 'org.gradle' since it conflicts with core plugins. You should use a different prefix for plugin: 'org.gradle.my-plugin.gradle.kts'. " +
```

Comment:
❓ Can we add the (full, or project-relative) path of the problematic script to the error message?

Reply:
Proposal for the error message:
```
Cannot define script plugin in the 'org.gradle' package: <path_of_script>

See https://docs.gradle.org/<version>/userguide/custom_plugins.html#sec:precompiled_plugins for more details.
```

I think that:
- we don't need to state the solution as it is obvious (use a different package 
- there's an explanation in the documentation.